tensorboard/plugins/debugger/tensor_helper.py

```diff
@@ -61,6 +62,54 @@ def parse_time_indices(s):
     return parsed[0]
 
 
+def translate_dtype(dtype):
+  """Translate numpy dtype into a string.
+
+  The 'object' type is understood as a tenosrflow string and translated into
+  'string'.
+
+  Args:
+    dtype: A numpy dtype object.
+
+  Returns:
+    A string representing the data type.
+  """
+  out = str(dtype)
+  # String-type TensorFlow Tensors are represented as object-type arrays in
+  # numpy. We map the type name back to 'string' for clarity.
+  return 'string' if out == 'object' else out
+
+
+def process_buffers_for_display(s, limit=40):
+  """Process a buffer for human-readable display.
+
+  This function performs the following operation on each of the buffers in `s`.
+    1. Truncate input buffer if the length of the buffer is greater than
```

Comment:
Maybe briefly offer some rationale on why we truncate in the first place.

Reply:
Done.